DatadogAlamofireExtension.podspec

```diff
@@ -14,6 +19,8 @@ Pod::Spec.new do |s|
     "Maciej Burda" => "maciej.burda@datadoghq.com"
   }
 
+  s.deprecated = true
```

Comment:
do we need to publish a version in order deprecate?

Can we simply delete this? So when customer tries to update (it is customer trigger action) they see an error and fix the issue instead keep using the deprecated version.

Reply:
We aim at deprecation not removal. The `DatadogAlamofireExtension` will continue to work for all customers in next releases until we finally delete it (in `3.0.0`).

The `deprecated = true` will produce a warning during `pod update` indicating that this pod is being sunsetted. Customers are expected to follow the notice from pod description and land on https://docs.datadoghq.com/real_user_monitoring/mobile_and_tv_monitoring/integrated_libraries/ios (available once https://github.com/DataDog/documentation/pull/24253 is merged) to instrument Alamofire using our `URLSessionInstrumentation` directly.